### ISSUE TITLE
build-stage0 fails on fedora

### ISSUE THREAD

**Issue by henriqueleng** [sabotage-linux/sabotage#355]

**Comment by rofl0r** [comment]
hmm maybe try adding -fPIC to the CFLAGS used by stage0_musl

**Comment by henriqueleng** [comment]
added 'echo "CFLAGS += -fPIC" >> config.mak' to pkg/stage0_musl, and it still the same error.
Am I doing something wrong?

**Comment by rofl0r** [comment]
can you post the build log somewhere ? pastebin oslt

**Comment by rofl0r** [comment]
the error is possibly due to a parrallel build issue
see http://git.musl-libc.org/cgit/musl/commit/?id=d18cf76d73df8f9cc751d4b4ba5a635c70c0c645
as an easy fix you could change the line ```make -j$MAKE_THREADS``` to ```make -j1```

**Comment by henriqueleng** [comment]
tried all this options and they didn't solved :(...
this is how  I let stage0_musl file: http://pastebin.com/raw.php?i=7axnQH1G
the error log(same than before): http://sprunge.us/RAJI

**Comment by richfelker** [comment]
Here is the source of the problem:
src/internal/libc.c:10: warning: visibility attribute not supported in this configuration; ignored

For building a shared libc.so, musl requires support for visibility. Originally there were workarounds for the case where the compiler doesn't support it, but they were removed since builds without visibility support would necessarily have ABI-breakage-type bugs anyway.

Sabotage is using gcc 3.4.6 for the bootstrap, right? This versions supports visibility just fine, but probably does something stupid to break it when --disable-shared is used.

**Comment by henriqueleng** [comment]
So should I try another compiler?

**Comment by henriqueleng** [comment]
Can you recommend me another way to get sabotage compiled and running?

**Comment by rofl0r** [comment]
yes, you can use musl-cross to "crosscompile" stage1 like its described in the readme. then you can use enter-chroot as usual to build stuff in the chroot.

**Comment by rofl0r** [comment]
(note that your problem is quite weird; build-stage0 builds gcc3.4.6 and then compiles musl with it, so the result should be 100% reproducible. may it be that the binutils version of your system causes the issue ? or did you set A to something else (like i386) when you actually wanted to build for x86_64 ?  could be helpful if you paste your config)

**Comment by henriqueleng** [comment]
I'm sure I didn't messed the config file, I retried the process lots of times. Any way, I got ubuntu and compiled it without problem, couldn't wait for the solution on fedora. The gcc version here is 4.8.4.
If it was a compiler problem, this problem may appear here again in the future.
